Sources/Harbor/HRequestManager.swift

```diff
@@ -87,8 +87,10 @@ extension HRequestManager {
             case 401:
                 if await !hasNewAuthorizationHeader(request: request) {
                     Self.config.authProvider?.authFailed()
+                    return .error(.authNeeded)
+                } else {
+                    return await self.request(model: model, request: request)
```

Comment:
This should take the amount of retries into account, doesn't it?

Reply:
No, because it is not related to the retries indicated in the request config. It should be executed if the request receives a 401 and new credentials are available.

Reply:
I think it should have a retry system, because it could have credentials, but they might not be enough (maybe the backend is failing, or simply the user has no access to that "resource/endpoint". We should avoid a loop

Reply:
A loop shouldn't happen in this case.  If the credentials change in the middle, it will retry only once.